packages/editor/src/schema.ts

```diff
@@ -0,0 +1,371 @@
+import type { Attrs, DOMOutputSpec, Mark, MarkSpec, NodeSpec, Node as PMNode } from 'prosemirror-model';
+
+export declare type EditorMarks =
+  | 'link'
+  | 'em'
+  | 'strong'
+  | 'code'
+  | 'underline'
+  | 'strikethrough'
+  | 'subscript'
+  | 'superscript';
+// | 'style';
+
+export declare type EditorNodes =
+  | 'doc'
+  | 'paragraph'
+  | 'blockquote'
+  | 'horizontalRule'
+  | 'heading'
+  | 'codeBlock'
+  | 'text'
+  | 'image'
+  | 'hardBreak'
+  | 'listItem'
+  | 'orderedList'
+  | 'bulletList';
+
+const SLOT = 0; // https://prosemirror.net/docs/guide/#schema.serialization_and_parsing
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export const isEmpty = (obj: Object): boolean => Object.keys(obj).length === 0;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const removeEntries = (obj: any, predicate: (key: string) => boolean): Record<string, string> => {
+  return (
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+    Object.keys(obj)
+      .filter(key => predicate(key))
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+      .reduce((acc, curr: any) => Object.assign(acc, { [curr]: obj[curr] }), {})
+  );
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const removeEmptyEntries = (obj: any): Record<string, string> => {
```

Comment:
Maybe here instead of `obj: any` we can use `obj: Record<string, unknown>` as well?